hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/cmd/UpdateCollectionIntervalsCommand.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2015-2017 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.agent.monitor.cmd;
+
+import java.util.Map;
+
+import org.hawkular.agent.monitor.extension.MonitorServiceConfiguration.AbstractEndpointConfiguration;
+import org.hawkular.agent.monitor.inventory.MonitoredEndpoint;
+import org.hawkular.agent.monitor.log.AgentLoggers;
+import org.hawkular.agent.monitor.log.MsgLogger;
+import org.hawkular.agent.monitor.protocol.EndpointService;
+import org.hawkular.agent.monitor.protocol.dmr.DMRNodeLocation;
+import org.hawkular.agent.monitor.protocol.dmr.DMRSession;
+import org.hawkular.bus.common.BasicMessageWithExtraData;
+import org.hawkular.bus.common.BinaryData;
+import org.hawkular.cmdgw.api.UpdateCollectionIntervalsRequest;
+import org.hawkular.cmdgw.api.UpdateCollectionIntervalsResponse;
+import org.hawkular.dmr.api.DmrApiException;
+import org.hawkular.dmr.api.OperationBuilder;
+import org.hawkular.dmr.api.OperationBuilder.CompositeOperationBuilder;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.client.ModelControllerClient;
+
+/**
+ * Update the specified metric and avail type collection intervals.  Performs a bulk update to prevent
+ * excessive agent restarts. Because metric types are not guaranteed to be consistent across agents, it is
+ * not a failure if a requested metric type does not exist.
+ *
+ * @author Jay Shaughnessy
+ */
+public class UpdateCollectionIntervalsCommand
```

Comment:
@theute I think there is a misunderstanding about that this new reload feature we introduced yesterday is doing. When an agent configuration setting changes now, ONLY THE AGENT SERVICE will restart. The entire app server does not restart - it should still be responsive to requests. It only affects the agent responsiveness (the agent won't be able to respond to websocket requests from the CFME UI, for example; and it will temporarily stop collecting metrics - but the restart really only takes a few seconds to complete).